src/link.d

```diff
@@ -773,21 +808,43 @@ version (Windows)
         cmd = toWinPath(cmd);
         version (CRuntime_Microsoft)
         {
-            if (strchr(cmd, ' '))
+            // Open scope so dmd doesn't complain about alloca + exception handling
             {
-                // MSVCRT: spawn does not work with spaces in the executable
-                size_t cmdlen = strlen(cmd);
-                char* shortName = (new char[](cmdlen + 1)).ptr; // enough space
-                uint plen = GetShortPathNameA(cmd, shortName, cast(uint)cmdlen + 1);
-                if (plen > 0 && plen <= cmdlen)
-                    cmd = shortName;
+                // Use process spawning through the WinAPI to avoid issues with executearg0 and spawnlp
+                OutBuffer cmdbuf;
+                cmdbuf.writestring("\"");
+                cmdbuf.writestring(cmd);
+                cmdbuf.writestring("\" ");
+                cmdbuf.writestring(args);
+
+                STARTUPINFOA startInf;
+                memset(&startInf, 0, STARTUPINFOA.sizeof);
```

Comment:
Documentation for STARTUPINFOA:
hStdInput
    If dwFlags specifies STARTF_USESTDHANDLES, this member is the standard input handle for the process. If STARTF_USESTDHANDLES is not specified, the default for standard input is the keyboard buffer.
<snip>
Otherwise, this member is ignored.